src/contrib/cluster/Akka.Cluster.Sharding/ShardRegion.cs

```diff
@@ -917,12 +983,90 @@ private void HandleShardRegionQuery(IShardRegionQuery query)
                     else
                         Sender.Tell(new ClusterShardingStats(ImmutableDictionary<Address, ShardRegionStats>.Empty));
                     break;
+                case GetEntityLocation g:
+                    ReplyToGetEntityLocationQuery(g, Sender);
+                    break;
                 default:
                     Unhandled(query);
                     break;
             }
         }
 
+        private void ReplyToGetEntityLocationQuery(GetEntityLocation getEntityLocation, IActorRef sender)
+        {
+            // Get the Address of the remote IActorRef, or return our Cluster.SelfAddress is the shard / entity
+            // is hosted locally.
+            Address GetNodeAddress(IActorRef shardOrRegionRef)
+            {
+                return shardOrRegionRef.Path.Address.HasGlobalScope
+                    ? shardOrRegionRef.Path.Address
+                    : Cluster.SelfAddress;
+            }
+
+            try
+            {
+                var shardId = ExtractShardId(new StartEntity(getEntityLocation.EntityId));
```

Comment:
One thing that is awkward about this query is that the `ShardExtractor` has to support the `ShardRegion.StartEntity` type in order for this to work - the reason being is that even though we already know what the entity id is, the extractors that resolve the `ShardId` are looking for an _input message_ to generate the `ShardId`, not the `EntityId`. Therefore I have to smuggle the `EnttityId` back through a `ShardRegion.StartEntity` message. That makes this a bit hacky but it's the best compromise I could make.